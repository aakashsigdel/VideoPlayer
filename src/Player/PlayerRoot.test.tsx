import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { create } from 'react-test-renderer'

import {
  PlayerRoot,
  IPlayerRootProps,
  DEFAULT_PLAYER_HEIGHT,
  DEFAULT_PLAYER_WIDTH,
} from './PlayerRoot'

describe('<PlayerRoot />', () => {
  let wrapper: ShallowWrapper<IPlayerRootProps>
  const defaultProps: IPlayerRootProps = {
    videoUrl: 'testUrl',
    posterUrl: 'testUrl',
  }

  beforeEach(() => {
    wrapper = shallow(<PlayerRoot {...defaultProps} />)
  })

  it('should render correctly', () => {
    const tree = create(<PlayerRoot {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should set height of the player to DEFAULT_PLAYER_HEIGHT when no height is specified as prop', () => {
    expect(wrapper.find('video').props().height).toBe(DEFAULT_PLAYER_HEIGHT)
  })

  it('should set width of the player to DEFAULT_PLAYER_WIDTH when no width is specified as prop', () => {
    expect(wrapper.find('video').props().width).toBe(DEFAULT_PLAYER_WIDTH)
  })
})
