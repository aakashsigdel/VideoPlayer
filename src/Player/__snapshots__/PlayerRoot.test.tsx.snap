// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerRoot /> should render correctly 1`] = `
<video
  controls={true}
  height={500}
  poster="testUrl"
  src="testUrl"
  width={620}
>
  Sorry your browser doesn't support HTML5 videos!
</video>
`;
